packages/android-client/FishjamClient/src/main/java/com/fishjamcloud/client/FishjamClient.kt

```diff
@@ -20,12 +20,19 @@ import com.fishjamcloud.client.webrtc.PeerConnectionManager
 import com.fishjamcloud.client.webrtc.RTCEngineCommunication
 import org.webrtc.Logging
 
-data class ConnectConfig(
+class ConnectConfig private constructor(
   val websocketUrl: String,
   val token: String,
-  val peerMetadata: Metadata,
-  val reconnectConfig: ReconnectConfig
-)
+  val reconnectConfig: ReconnectConfig,
+  val peerMetadata: Metadata
+) {
+  public constructor(
+    url: String,
+    token: String,
+    peerMetadata: Metadata,
+    reconnectConfig: ReconnectConfig
+  ) : this(websocketUrl = "$url/socket/peer/websocket", token = token, peerMetadata = peerMetadata, reconnectConfig = reconnectConfig) {}
+}
```

Comment:
What could be better way to handle this logic in Kotlin? 🤔

Reply:
```
data class ConnectConfig(
  private _websocketUrl: String,
  val token: String,
  val peerMetadata: Metadata,
  val reconnectConfig: ReconnectConfig
){
 val websocketUrl : String
    get() = "$_websocketUrl/socket/peer/websocket"
}
```

Something like this should work ;)